module/support/optimisation/NSGA2Evaluator/src/tasks/EvaluatorTask.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef MODULE_SUPPORT_OPTIMISATION_EVALUATORTASK_H
+#define MODULE_SUPPORT_OPTIMISATION_EVALUATORTASK_H
+
+#include <nuclear>
+#include <vector>
+
+#include "NSGA2Evaluator.hpp"
+
+#include "message/platform/RawSensors.hpp"
+#include "message/platform/webots/messages.hpp"
+#include "message/support/optimisation/NSGA2Evaluator.hpp"
+#include "message/support/optimisation/NSGA2Optimiser.hpp"
+
+namespace module::support::optimisation {
+    using message::platform::RawSensors;
+    using message::platform::webots::OptimisationRobotPosition;
+    using message::support::optimisation::NSGA2EvaluationRequest;
+    using message::support::optimisation::NSGA2FitnessScores;
+
+    class NSGA2Evaluator;  // forward declaration
+
+    class EvaluatorTask {
+    public:
+        virtual void process_raw_sensor_msg(const RawSensors& sensors, NSGA2Evaluator* evaluator)   = 0;
```

Comment:
this should be a smart pointer

Reply:
also these should all have documentation directives, explaining generically what the functions should mean